ls1intum/bamboo-server-notification-plugin
# [Feature] Add GCC as SCA tool for C

### Checklist

### Motivation and Context


Adds support for GCC as a static analysis tool. A feature the operating systems lecture would like to use next semester.

### Description

I added an additional artifact called `GCC` that will be also used in the _static-code-analysis-parser_ and _Artemis_ itself.

### Steps for Testing


1. Replace the dependency for the SCA parser e.g. `static-code-analysis-parser-1.3.4.jar` with the one in the [related branch](https://github.com/ls1intum/static-code-analysis-parser/tree/feature/add-c-language-support) that can be obtained by running `atlas-package` in the parsers root directory.
2. Deploy the plugin on a (local) on Bamboo.
3. Run the [related Artemis branch](https://github.com/ls1intum/Artemis/tree/feature/add-c-sca-support) (locally).
4. Navigate to Artemis and create a C programming exercise with SCA enabled. 
5. Create some issue like a double free error in the code, then check out the result in Artemis.
